.pipelines/templates/add-windows-nodepool-job.yaml

```diff
@@ -0,0 +1,62 @@
+parameters:
+  depend: ""
+  clusterName: "" # unique identifier
+  vmSize: ""
+  retryCount: 5
+
+jobs:
+- job: windows_nodepool
+  displayName: Add Windows Nodepool
+  dependsOn: ${{ parameters.depend }}
+  pool:
+    name: $(BUILD_POOL_NAME_DEFAULT)
+    demands:
+    - agent.os -equals Linux
+    - Role -equals $(CUSTOM_E2E_ROLE)
+  timeoutInMinutes: 30
+  steps:
+    - task: AzureCLI@2
+      inputs:
+        azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+        scriptLocation: "inlineScript"
+        scriptType: "bash"
+        addSpnToEnvironment: true
+        inlineScript: |
+          set -e
+          
+          windows_nodepool=$(az aks nodepool list \
+          --resource-group ${{ parameters.clusterName }} \
+          --cluster-name ${{ parameters.clusterName }} \
+          --query "[?osType=='Windows']" \
+          --output tsv)
+
+          if [ -z "$windows_nodepool" ]; then
+            echo "No Windows node pool found in the AKS cluster."
+
+            # wait for cluster to update
+            while true; do
+              cluster_state=$(az aks show \
+              --name "${{ parameters.clusterName }}" \
+              --resource-group "${{ parameters.clusterName }}" \
+              --query provisioningState)
+              
+              if echo "$cluster_state" | grep -q "Updating"; then
+                echo "Cluster is updating. Sleeping for 30 seconds..."
+                sleep 30
+              else
+                break
+              fi
```

Comment:
We should output `$cluster_state` once we are complete with the loop. The current way the logic is setup is only capturing if the state is of `Updating*`. In the future we may want to capture other states and it would be nice to know which one just by looking at the logs.